### PR TITLE
add http_host and tls_sni fields to ClientInfo

### PR DESCRIPTION
**Pull request by bharatnc**
### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `http_host` and `tls_sni` fields to ClientInfo.

Adds `http_host` and `tls_sni` fields to ClientInfo and also populates the values into the relevant system tables:
```
➜  ~ curl 'http://localhost:8123/?query=SELECT%201'
:) select query, event_date, event_time, http_host, tls_sni from system.query_log where query_id = '2018213a-7129-400f-b133-01cf3a7d6a23' FORMAT Vertical;

Row 1:
──────
query:      SELECT 1
event_date: 2022-06-29
event_time: 2022-06-29 19:25:32
http_host:  localhost:8123
tls_sni:    127.0.0.1:58522
```

Related to: #38323

### CODE REVIEW

File: src/Interpreters/ClientInfo.cpp [hunk 1]
```diff
@@ -56,6 +56,12 @@ void ClientInfo::write(WriteBuffer & out, UInt64 server_protocol_revision) const
 
         if (server_protocol_revision >= DBMS_MIN_REVISION_WITH_REFERER_IN_CLIENT_INFO)
             writeBinary(http_referer, out);
+
+        if (server_protocol_revision >= DBMS_MIN_REVISION_WITH_REFERER_IN_CLIENT_INFO)
```
Comment (alexey-milovidov) [inline src/Interpreters/ClientInfo.cpp:60]: A new macro should be used for compatibility.

**Comment by bharatnc** [comment]
Related failures don't seem to be related, should be ready for review.

File: src/Core/ProtocolDefines.h [hunk 2]
```diff
@@ -45,6 +45,10 @@
 
 #define DBMS_MIN_REVISION_WITH_CUSTOM_SERIALIZATION 54454
 
+#define DBMS_MIN_REVISION_WITH_HTTP_HOST_IN_CLIENT_INFO 54450
```
Comment (CheSema) [inline src/Core/ProtocolDefines.h:48]: I have concerns about that numbers. I see that VERSION_REVISION is defined as 54465 now.
I'm verifying that part right now. I'll return soon.

**Review by CheSema — CHANGES_REQUESTED** [review]
Please fix the flaws in the comments above.
You need to extract information from ssl layer for tls_sni.

**Comment by bharatnc** [comment]
> Please fix the flaws in the comments above. You need to extract information from ssl layer for tls_sni.

Ok sure I will continue with this PR.

**Comment by robot-clickhouse** [comment]
This is an automated comment for commit 318663e01a7ed3e8712b316ad5805730884fdfec with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/38581/318663e01a7ed3e8712b316ad5805730884fdfec/ci_running.html)
The overall status of the commit is 🔴 failure

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>Docs Check</td><td>Builds and tests the documentation</td><td>🔴 failure</td></tr>
</table>

**Comment by bharatnc** [comment]
> > Please fix the flaws in the comments above. You need to extract information from ssl layer for tls_sni.
> 
> Ok sure I will continue with this PR.

Sorry for the delay. I will get back to work on this.

**Comment by bharatnc** [comment]
Sorry that this PR has been idle for a very long time. I'm going to close this PR for now since I've not had the time to work on the tls sni bit and the PR has been piling up merge conflicts. 

I've tried adding the sni support on the Poco secure socket implementation side, but was unsuccessful in writing a integration test using tls sni setup correctly. I will try reopening it or will open a new PR if I manage to get it working.

**Pull request closed without merging**